src/XRP/default-xrp-client.ts

```diff
@@ -466,4 +425,70 @@ export default class DefaultXrpClient implements XrpClientDecorator {
     )
     return XrpTransaction.from(getTransactionResponse, this.network)
   }
+
+  /**
+   * Populates the required fields common to all transaction types.
+   * @see https://xrpl.org/transaction-common-fields.html
+   *
+   * Note: The returned Transaction object must still be assigned transaction-specific details.
+   *
+   * @param wallet The wallet that will sign and submit this transaction.
+   * @returns A promise which resolves to a Transaction protobuf with the required common fields populated.
+   */
+  private async prepareBaseTransaction(wallet: Wallet): Promise<Transaction> {
+    const classicAddress = Utils.decodeXAddress(wallet.getAddress())
+    if (!classicAddress) {
+      throw XrpError.xAddressRequired
+    }
+    const fee = await this.getMinimumFee()
+    const accountData = await this.getAccountData(classicAddress.address)
+    const openLedgerSequence = await this.getOpenLedgerSequence()
+
+    const senderAccountAddress = new AccountAddress()
+    senderAccountAddress.setAddress(wallet.getAddress())
+
+    const account = new Account()
+    account.setValue(senderAccountAddress)
+
+    const lastLedgerSequence = new LastLedgerSequence()
+    lastLedgerSequence.setValue(openLedgerSequence + maxLedgerVersionOffset)
+
+    const signingPublicKeyBytes = Utils.toBytes(wallet.publicKey)
+    const signingPublicKey = new SigningPublicKey()
+    signingPublicKey.setValue(signingPublicKeyBytes)
+
+    const transaction = new Transaction()
+    transaction.setAccount(account)
+    transaction.setFee(fee)
+    transaction.setSequence(accountData.getSequence())
+    transaction.setLastLedgerSequence(lastLedgerSequence)
+    transaction.setSigningPublicKey(signingPublicKey)
+
+    return transaction
+  }
+
+  /**
+   * Signs the provided Transaction object using the provided Wallet and submits to the XRPL network.
+   *
+   * @param wallet The wallet that will sign and submit this transaction.
+   * @returns A promise which resolves to a string representing the hash of the submitted transaction.
+   */
+  private async submitTransaction(
+    wallet: Wallet,
+    transaction: Transaction,
```

Comment:
Optional / This is a very subjective comment: I'd consider reversing this parameter order so that it reads more logically:
"submit transaction `transaction` (from) `wallet`"

rather than:
"submit transaction (with) `wallet` (signing) `transaction`"

Reply:
I agree with this